CONTRIBUTING.md

```diff
@@ -28,3 +28,49 @@ Members of the Contributors team are encouraged to review pull requests that
 have already been reviewed, and pull request contributors are encouraged to seek
 multiple reviews.  Reviews from anyone not on the Contributors team are always
 appreciated and encouraged!
+
+## Reviewing Pull Requests
+
+All mlpack contributors who choose to review and provide feedback on pull
+requests have a responsibility to both the project and the individual making
+the contribution. 
+
+Reviews and feedback should be
+[helpful, insightful, and geared towards improving the contribution](
+  https://www.youtube.com/watch?v=NNXk_WJzyMI).
+If there are reasons why you feel the PR should not be merged, explain
+what those are. Be open to having your mind changed. Be open to
+working with the contributor to make the pull request better.
+
+Please don't leave dismissive or disrespectful reviews!  It's not helpful for
+anyone.
+
+When reviewing a pull request, the primary goals are:
+
+- For the codebase/project to improve
+- For the person submitting the request to succeed
+
+Even if a pull request does not get merged, the submitters should come away
+from the experience feeling like their effort was not wasted or unappreciated.
+Every pull request from a new contributor is an opportunity to grow the community. 
+
+When changes are necessary, request them, do not demand them, and do not assume
+that the contributor already knows how to do that. Be there to lend a helping
+hand in case of need.
+
+Since there can sometimes be a lot more pull requests being opened than
+reviewed, we highly encourage everyone to review each others pull request
+keeping in mind all the above mentioned points.
+
+Let's welcome new contributors with ❤️.
+
+## Pull Request Waiting Time 
+
+mlpack is a community-driven project, so everyone only works on it in their
+free time, this means it may take some time for them to review pull requests.
```

Comment:
```suggestion
free time; this means it may take some time for them to review pull requests.
```
This one is really pedantic. :smile:

Reply:
Actually I guess I'll just commit it because there have already been a couple approvals, so I can merge the whole PR just afterwards. :)